cpp/dolfinx/refinement/option.h

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2024 Paul Kühner
+//
+// This file is part of DOLFINX (https://www.fenicsproject.org)
+//
+// SPDX-License-Identifier:    LGPL-3.0-or-later
+
+#pragma once
+
+#include <cstdint>
+#include <type_traits>
+
+namespace dolfinx::refinement
+{
+
+/// @brief Options for data to compute during mesh refinement.
+enum class Option : std::uint8_t
+{
+    none = 0b00,
```

Comment:
Add comments (doxygen compatible) for each option

Reply:
Addressed in #3322 with 135ef1f60bf37edb26291ac03db3830ec1e0b00d. Which needs to merged prior to this one.